src/tests/FunctionLookupTest.cpp

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "../FunctionLookup.h"
+#include "iostream"
+#include <gtest/gtest.h>
+
+using namespace io::substrait;
+
+class VeloxFunctionMappings : public FunctionMapping {
+public:
+  static const std::shared_ptr<VeloxFunctionMappings> make() {
+    return std::make_shared<VeloxFunctionMappings>();
+  }
+  /// scalar function names in difference between  velox and Substrait.
+  const FunctionMap scalaMapping() const override {
+    static const FunctionMap scalarMappings{
+        {"plus", "add"}, {"minus", "subtract"}, {"mod", "modulus"},
+        {"eq", "equal"}, {"neq", "not_equal"},  {"substr", "substring"},
+    };
+    return scalarMappings;
+  };
+};
+
+class SubstraitFunctionLookupTest : public ::testing::Test {
+protected:
+  void SetUp() override {
+    ExtensionPtr extension_ = Extension::load();
+    FunctionMappingPtr mappings_ =
+        std::make_shared<const VeloxFunctionMappings>();
+    scalarFunctionLookup_ =
+        std::make_shared<FunctionLookup>(extension_, mappings_);
+    aggregateFunctionLookup_ =
+        std::make_shared<FunctionLookup>(extension_, mappings_);
+  }
+
+  void testScalarFunctionLookup(const std::string &name,
+                                const std::vector<TypePtr> &arguments,
+                                const std::string &outputSignature) {
+    const auto &functionVariant =
+        scalarFunctionLookup_->lookupScalarFunction(name, arguments);
```

Comment:
Cider not only need comparison of input type, but also need comparison of output type when looking up function.

Reply:
reverted